src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritConnection.java

```diff
@@ -445,7 +448,9 @@ public void run() {
                     if (!channel.isConnected() || !sshConnection.isConnected()) {
                         throw new IllegalStateException("SSH connection is already lost.");
                     }
-                    sleep(SSH_RX_SLEEP_MILLIS);
+                    if (linecount > 0) {
```

Comment:
Not sure if this a good idea. The thing is if we get an event that is 10 times the size of the buffer, it would take 1 sec to read this event in. On the other hand if for some reason the reader.read is not blocking in case no data is there or it can't fill the buffer (like in the bug we try to fix) we would consume a lot of CPU time in this thread.